systemd/emulation.go

```diff
@@ -98,12 +99,22 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", fmt.Errorf("bind-mounted directory is not supported in emulation mode")
 	}
 
-	mountUnitName, actualFsType, options, err := writeMountUnitFile(snapName, revision, what, where, fstype)
+	// In emulation mode actualFsType is the fs we want to use to manually mount
+	// the snap below, but fstype is used for the created mount unit.
+	// This means that when preseeding in a lxd container, the snap will be
+	// mounted with fuse, but mount unit will use squashfs.
+	actualFsType, actualOptions, err := actualFs(fstype)
```

Comment:
afaict this code now can move past writeMountUnitFile because the result is only used in the command?